### PR TITLE
[image-manipulator] Introduce a new API for image manipulation on Web

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Following up on #30135

# How

Implemented the new API for `expo-image-manipulator` on Web, added tests to test-suite and fixed a few small things

# Test Plan

For both legacy and the new API, NCL examples and tests work fine

### CODE REVIEW

File: apps/test-suite/tests/ImageManipulator.js
Comment (tsapeta) [inline apps/test-suite/tests/ImageManipulator.js:1]: This has been renamed to `*.ts`, but git shows it as deleted 🤷

File: apps/test-suite/tests/ImageManipulator.ts
Comment (tsapeta) [inline apps/test-suite/tests/ImageManipulator.ts:1]: This file now contains equivalent tests but for the new API, the previous one are described as `ImageManipulator (Legacy)`

File: packages/expo-image-manipulator/src/web/utils.web.ts
Comment (tsapeta) [inline packages/expo-image-manipulator/src/web/utils.web.ts:1]: Everything was just moved from `ExpoImageManipulator.web.ts` to here

File: packages/expo-modules-core/src/web/CoreModule.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ class SharedObject<TEventsMap extends Record<never, never>>
   implements SharedObjectType
 {
   release(): void {
-    throw new Error('Method not implemented.');
+    // no-op on Web, but subclasses can override it if needed.
```
Comment (tsapeta) [inline packages/expo-modules-core/src/web/CoreModule.ts:98]: We shouldn't throw from here by default as some methods (especially hooks like `useReleasingSharedObject`) can call this and catching the errors doesn't seem reasonable

**Review by Kudo — APPROVED** [review]

**Pull request merged**